pkgs/development/embedded/bootcommander/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchgit, cmake, libopenblt }:
+
+stdenv.mkDerivation rec {
+
+  pname = "bootcommander";
+  version = "011500";
+
+  src = fetchgit {
+    url = "https://github.com/feaser/openblt.git";
+    rev = "openblt_v${version}";
+    sparseCheckout = [ "Host/Source/" ];
+    sha256 = "sha256-cWdc6Imd6IBqYQRNEafp3bDDQLgdpE90fxSkY+vzhAM=";
+  };
+
+  preConfigure = "cd ./Host/Source/BootCommander/";
```

Comment:
We should use sourceRoot or similar

Reply:
I can't use sourceroot because `make` saves the binary output in different locations that otherwise would not be accessible with sourceroot.

Reply:
I tried over and over to get this thing to work and I can't. I need help @SuperSandro2000 @fgaz 
Since the CMakeList.txt file uses fixed paths for the executables ouptput dirs (https://github.com/feaser/openblt/blob/0e1104d711c3993cf83f47f7a6bef3485efeb3dc/Host/Source/LibOpenBLT/CMakeLists.txt#L71) I can't use `sourceRoot="source/Host/Source/BootCommander"` otherwise the linker fails with error. The reason is that the compilation works but the linker tries to output the file to the directory `../../../` that is out of scope in the nix sandbox build dir. The same values that control this behaviour (`PROJECT_OUTPUT_DIRECTORY,CMAKE_RUNTIME_OUTPUT_DIRECTORY`...) can't be overridden in `cmakeFlags` because they will not be read since cmake has precedence on the actual ones defined in the CMakeList.txt instead of the command line ones. So the last options are either to patch the cmake file or to just cd in to the folder and build from there. Otherwise I don't know how to build this program.